lib/rules/no-constant-condition.js

```diff
@@ -132,7 +138,13 @@ module.exports = {
         return {
             ConditionalExpression: reportIfConstant,
             IfStatement: reportIfConstant,
-            WhileStatement: checkLoop,
+            WhileStatement(node) {
+                if (node.test && node.test.raw === "true" && checkLoops === "allExceptWhileTrue") {
```

Comment:
```suggestion
                if (node.test.type === "Literal" && node.test.value === true && checkLoops === "allExceptWhileTrue") {
```

It's preferrable to check the AST.

Reply:
Done!